### DISCUSSION TITLE
help please physx does not compile on linux

### DISCUSSION THREAD

**Discussion by MantraxStudios** [NVIDIA-Omniverse/PhysX#295] [Q&A]
Hoal I am trying to compile physx on linux but it gives me these errors attached photo I hope you can help me thank you

![Screenshot from 2024-07-11 06-54-40](https://github.com/NVIDIA-Omniverse/PhysX/assets/85538374/1e40a586-24e1-400a-996c-100f67b60b1c)

**Comment by vreutskyy** [comment]
Hi @GrstudiosGamesPro 
Perhaps you may try adding the `-Wno-unsafe-buffer-usage` compiler option to silence that warning.

**Reply by MantraxStudios** [reply]
I didn't do it but I am formatting the pc 🥺

**Reply by MantraxStudios** [reply]
i already formatted the pc but i got this x11.h error

**Reply by MantraxStudios** [reply]
![image](https://github.com/user-attachments/assets/d1315550-54d0-4fb3-b8fe-14663fe226d1)

**Reply by MantraxStudios** [reply]
![image](https://github.com/user-attachments/assets/3e9a8101-33d9-4cc8-9108-cadf3aa0e924)

**Reply by vreutskyy** [reply]
Sorry, not a Linux expert, but quick googling shows you may need to install x11 library
https://stackoverflow.com/questions/5299989/x11-xlib-h-not-found-in-ubuntu

It doesn't look like PhysX side problem to me, but again, I mostly use Windows.

On the second screen, your engine seems to miss some PhysX's libs linked.

**Reply by preist-nvidia** [reply]
@GrstudiosGamesPro - please have a look at the linux platform readme - and we can only provide support for the configurations that we list in this doc: https://github.com/NVIDIA-Omniverse/PhysX/blob/main/physx/documentation/platformreadme/linux/README_LINUX.md
